conformance/lang/expressions/multiplicative-expr/division.balt

```diff
@@ -0,0 +1,4506 @@
+
+    v = j / c;
+    io:println(v.toBalString()); // @output ()
+
+    v = j / d;
+    io:println(v.toBalString()); // @output ()
+
+    v = j / e;
+    io:println(v.toBalString()); // @output ()
+
+    v = j / f;
+    io:println(v.toBalString()); // @output ()
+
+    v = j / g;
+    io:println(v.toBalString()); // @output ()
+
+    v = j / h;
+    io:println(v.toBalString()); // @output ()
+
+    v = j / i;
+    io:println(v.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test integer division where at least one operand is of an int subtype union.
+Fail-Issue: ballerina-platform/ballerina-lang#32912
+Labels: multiplicative-expr, int, DecimalNumber, byte, int:Signed8, int:Signed16,
+        int:Signed32, int:Unsigned8, int:Unsigned16, int:Unsigned32, module-type-defn, union-type
+
+type T1 1|2;
+
+type T2 int:Signed16|int:Signed32|int:Unsigned8|int:MAX_VALUE;
+
+type T3 int:Unsigned16|int:Unsigned32|byte|int:Signed8;
+
+type T4 T1|T2;
+
+public function main() {
+    T1 a = 1;
+    T2 b = 2;
+    T3 c = 3;
+    T4 d = 4;
+
+    int v = 2 / a;
+    io:println(v); // @output 2
+
+    v = 2 / b;
+    io:println(v); // @output 1
+
+    v = 2 / c;
+    io:println(v); // @output 0
+
+    v = 2 / d;
+    io:println(v); // @output 0
+
+    v = a / a;
+    io:println(v); // @output 1
+
+    v = a / b;
+    io:println(v); // @output 0
+
+    v = a / c;
+    io:println(v); // @output 0
+
+    v = a / d;
+    io:println(v); // @output 0
+
+    v = b / b;
+    io:println(v); // @output 1
+
+    v = b / c;
+    io:println(v); // @output 0
+
+    v = b / d;
+    io:println(v); // @output 0
+
+    v = c / c;
+    io:println(v); // @output 1
+
+    v = c / d;
+    io:println(v); // @output 0
+
+    v = d / d;
+    io:println(v); // @output 1
+}
+
+Test-Case: output
+Description: Test integer division with optional int operand's type written in different ways.
+Labels: multiplicative-expr, int, DecimalNumber, nil-literal, optional-type, module-type-defn, union-type, value:toBalString
+
+type T1 1|()|2;
+
+type T2 1|2?;
+
+public function main() {
+    T1 a = 1;
+    T2 b = 2;
+    ()|int c = ();
+
+    int? v = a / a;
+    io:println(v); // @output 1
+
+    v = a / b;
+    io:println(v); // @output 0
+
+    v = a / c;
+    io:println(v.toBalString()); // @output ()
+
+    v = b / b;
+    io:println(v); // @output 1
+
+    v = b / c;
+    io:println(v.toBalString()); // @output ()
+
+    v = c / c;
+    io:println(v.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test integer division with HexIntLiterals.
+Labels: multiplicative-expr, int, DecimalNumber, HexIntLiteral
+
+public function main() {
+    int a = 0xdef;
+    int b = a / 0x12;
+    io:println(b); // @output 198
+    io:println(a / 18); // @output 198
+    io:println(0Xdef / 18); // @output 198
+}
+
+Test-Case: output
+Description: Test division scenarios for different decimal values.
+Labels: multiplicative-expr, numeric-literal, float-point-literal
```

Comment:
Shall we use `DecimalFloatingPointNumber` instead of `floating-point-literal`?
Don't need `numeric-literal` label? 
Missing `decimal`, `DecimalTypeSuffix` labels.